commit 8423f730a75b1fe718c45a9bf9ca03b87fb96f0e (refs/changes/48/1048/3)
Author: Jerry Jelinek <jerry.jelinek@joyent.com>
Date:   2016-12-02T20:39:07+00:00 (2 years, 10 months ago)
    
    OS-5826 Disk layout server setup parameters not taking effect
    Reviewed by: Orlando Vazquez <orlando@joyent.com>
    Reviewed by: Josh Wilsdon <jwilsdon@joyent.com>
    Reviewed by: Pedro Palazon Candel <pedro@joyent.com>
    Approved by: Orlando Vazquez <orlando@joyent.com>

diff --git a/overlay/generic/lib/sdc/config.sh b/overlay/generic/lib/sdc/config.sh
index 782e7725..0c8e5621 100644
--- a/overlay/generic/lib/sdc/config.sh
+++ b/overlay/generic/lib/sdc/config.sh
@@ -7,7 +7,7 @@
 #
 # bash config.sh -json
 #
-# Copyright (c) 2015 Joyent Inc., All rights reserved.
+# Copyright 2016 Joyent Inc.
 #
 
 CACHE_FILE_JSON="/tmp/.config.json"
@@ -73,7 +73,9 @@ function load_sdc_sysinfo {
     eval $(/usr/bin/sysinfo -p | sed -e "s/^/${prefix}/")
 }
 
-# Sets SDC_CONFIG_FILENAME with the location of the USB config file, or /opt/smartdc/config
+# Sets SDC_CONFIG_FILENAME with the location of the config file. This can
+# come from the USB key, /opt/smartdc/config/node.config, or (if on an unsetup
+# CN) /var/tmp/node.config/node.config.
 function load_sdc_config_filename {
 
     # the default
@@ -87,6 +89,10 @@ function load_sdc_config_filename {
 
         if [[ -f ${SDC_CONFIG_FILENAME} ]]; then
             SDC_CONFIG_INC_DIR="$(dirname ${SDC_CONFIG_FILENAME})/config.inc"
+        elif [[ ! -f ${COMPUTE_NODE_CONFIG_FILENAME} ]]; then
+            if [[ -f /var/tmp/node.config/node.config ]]; then
+                COMPUTE_NODE_CONFIG_FILENAME=/var/tmp/node.config/node.config
+            fi
         fi
 
         if [[ -f ${COMPUTE_NODE_CONFIG_FILENAME} ]]; then
